pymeasure/instruments/keithley/keithley6517b.py

```diff
@@ -0,0 +1,343 @@
+        super(Keithley6517B, self).__init__(
+            adapter, "Keithley 6517B Electrometer/High Resistance Meter",
+            **kwargs
+        )
+
+    def enable_source(self):
+        """ Enables the source of current or voltage depending on the
+        configuration of the instrument. """
+        self.write("OUTPUT ON")
+
+    def disable_source(self):
+        """ Disables the source of current or voltage depending on the
+        configuration of the instrument. """
+        self.write("OUTPUT OFF")
+
+    def measure_resistance(self, nplc=1, resistance=2.1e5, auto_range=True):
+        """ Configures the measurement of resistance.
+
+        :param nplc: Number of power line cycles (NPLC) from 0.01 to 10
+        :param resistance: Upper limit of resistance in Ohms,
+                           from -210 POhms to 210 POhms
+        :param auto_range: Enables auto_range if True, else uses the
+                           resistance_range attribut
+        """
+        log.info("%s is measuring resistance.", self.name)
+        self.write(":SENS:FUNC 'RES';"
+                   ":SENS:RES:NPLC %f;" % nplc)
+        if auto_range:
+            self.write(":SENS:RES:RANG:AUTO 1;")
+        else:
+            self.resistance_range = resistance
+        self.check_errors()
+
+    def measure_voltage(self, nplc=1, voltage=21.0, auto_range=True):
+        """ Configures the measurement of voltage.
+
+        :param nplc: Number of power line cycles (NPLC) from 0.01 to 10
+        :param voltage: Upper limit of voltage in Volts, from -1000 V to 1000 V
+        :param auto_range: Enables auto_range if True, else uses the
+                           voltage_range attribut
+        """
+        log.info("%s is measuring voltage.", self.name)
+        self.write(":SENS:FUNC 'VOLT';"
+                   ":SENS:VOLT:NPLC %f;" % nplc)
+        if auto_range:
+            self.write(":SENS:VOLT:RANG:AUTO 1;")
+        else:
+            self.voltage_range = voltage
+        self.check_errors()
+
+    def measure_current(self, nplc=1, current=1.05e-4, auto_range=True):
+        """ Configures the measurement of current.
+
+        :param nplc: Number of power line cycles (NPLC) from 0.01 to 10
+        :param current: Upper limit of current in Amps, from -21 mA to 21 mA
+        :param auto_range: Enables auto_range if True, else uses the
+                           current_range attribut
+        """
+        log.info("%s is measuring current.", self.name)
+        self.write(":SENS:FUNC 'CURR';"
+                   ":SENS:CURR:NPLC %f;" % nplc)
+        if auto_range:
+            self.write(":SENS:CURR:RANG:AUTO 1;")
+        else:
+            self.current_range = current
+        self.check_errors()
+
+    def auto_range_source(self):
+        """ Configures the source to use an automatic range.
+        """
+        self.write(":SOUR:VOLT:RANG:AUTO 1")
+
+    def apply_voltage(self, voltage_range=None,
```

Comment:
Is this methods incompletely migrated from another instrument? Both `self.source_mode` and `self.compliance_current` are not defined in the present implementation, and are only defined in here.